app/v2/controllers/CreateAmendForeignPropertyAnnualSubmissionController.scala

```diff
@@ -57,74 +54,56 @@ class CreateAmendForeignPropertyAnnualSubmissionController @Inject() (val authSe
 
   def handleRequest(nino: String, businessId: String, taxYear: String): Action[JsValue] =
     authorisedAction(nino).async(parse.json) { implicit request =>
-      implicit val correlationId: String = idGenerator.getCorrelationId
-      logger.info(
-        message = s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] " +
-          s"with correlationId : $correlationId")
-      val rawData = CreateAmendForeignPropertyAnnualSubmissionRawData(nino, businessId, taxYear, request.body)
-      val result =
-        for {
-          parsedRequest   <- EitherT.fromEither[Future](parser.parseRequest(rawData))
-          serviceResponse <- EitherT(service.createAmendForeignPropertyAnnualSubmission(parsedRequest))
-        } yield {
-          val hateoasData    = CreateAmendForeignPropertyAnnualSubmissionHateoasData(nino, businessId, taxYear)
-          val vendorResponse = hateoasFactory.wrap(serviceResponse.responseData, hateoasData)
+      implicit val ctx: RequestContext = RequestContext.from(idGenerator, endpointLogContext)
 
-          logger.info(
-            s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] - " +
-              s"Success response received with CorrelationId: ${serviceResponse.correlationId}")
+      val rawData = CreateAmendForeignPropertyAnnualSubmissionRawData(nino, businessId, taxYear, request.body)
 
-          val response = Json.toJson(vendorResponse)
+      val requestHandler = RequestHandler
+        .withParser(parser)
+        .withService(service.createAmendForeignPropertyAnnualSubmission)
+        .withAuditing(
+          auditHandler(rawData = rawData, correlationId = ctx.correlationId, nino = nino, businessId = businessId, taxYear = taxYear, request = request))
+        .withHateoasResult(hateoasFactory)(CreateAmendForeignPropertyAnnualSubmissionHateoasData(nino, businessId, taxYear), OK)
 
-          auditSubmission(GenericAuditDetail(request.userDetails, rawData, serviceResponse.correlationId, AuditResponse(OK, Right(Some(response)))))
+      requestHandler.handleRequest(rawData)
+    }
 
-          Ok(response)
-            .withApiHeaders(serviceResponse.correlationId)
+  private def auditHandler(rawData: CreateAmendForeignPropertyAnnualSubmissionRawData,
```

Comment:
same thing about passing in the raw data as in the other controller... i imagine this is a recurring thing so worth checking through all the controllers you've updated to see if its the same case